### ISSUE TITLE
Interactive rebase over tramp no longer brings up editor to edit commit log

### ISSUE THREAD

**Issue by Jehops** [magit/magit#4542]
Recipe:

- In the `magit-status` buffer for a repository opened over tramp, hit `r` `i` on a commit to rebase interactively.
- The `rebase-todo` buffer is opened normally.
- Make changes such as hitting `w` to reword a commit.
- Hit `C-c C-c`.
- The problem is at this stage.  A buffer is no longer opened to edit the commit message.  The only workaround that I have found is to go to a shell prompt on the remote host and run `git rebase --abort`. 

`M-x toggle-debug-on-error RET` does not report any errors.  The `magit-process` buffer says
```
run git … rebase -i --autostash 0fbe8c6b6bce\^
hint: Waiting for your editor to close the file...
WITH-EDITOR: 73924 OPEN /usr/ports/.git/rebase-merge/git-rebase-todo IN /usr/ports
hint: Waiting for your editor to close the file...
WITH-EDITOR: 74073 OPEN /usr/ports/.git/COMMIT_EDITMSG IN /usr/ports
```

The repository is opened at `/ssh:<remote_host>:/usr/ports/`.

I tried this with the latest melpa package as well as the latest from the repository with `make emacs-Q`, which says `Magit v3.3.0-29-g2bd1d823, Git 2.33.1, Emacs 29.0.50, berkeley-unix`.

**Comment by tarsius** [comment]
Please update `with-editor`, restart Emacs and try again.

**Comment by Jehops** [comment]
`with-editor` from melpa was already at the latest version `20211028.2105`.  I reinstalled it, but the result is the same.

**Comment by tarsius** [comment]
> The problem is at this stage. A buffer is no longer opened to edit the commit message.

The instructions are ambiguous. According to the previous steps the "commit message buffer" hasn't even been shown yet. You were in the "list of commits to be rebased buffer" and then typed `C-c C-c`. Are you trying to say that after doing that the "commit message buffer" is not created (which is very different from "once it existed but now it doesn't exist anymore").

I am able to use rebase to change a commit message over tramp.

**Comment by Jehops** [comment]
Apologies if I'm unclear.   Let me try again.

From the `git-rebase-todo` buffer (example contents of the buffer are below) when I choose `r` to reword or `s` to sqush commits, then hit `C-c C-c` to make it happen, in the past the `COMMIT_EDITMSG` buffer was created and displayed so that I could edit the commit message.  Now, if the repository that I'm working is remote (e.g., `/ssh:<remote_host>:/usr/ports/`) then `COMMIT_EDITMSG` is not created.  If I ssh to the remote host then open emacs from the remote host, this problem does not occur.  That is, the `COMMIT_EDITMSG` buffer is created and displayed and I can edit the commit log.

Is that clear?  If not, I can create a video if that's helpful.

```
pick 97e29f81cf3d Mk/Uses/emacs.mk: Do not use Emacs major version in package name
squash 9297f0d50c3f Mk/Uses/emacs.mk: Do not use Emacs major version in package names
pick 221562890ac3 Fix PORTREVISON bumps based on yasu's investigation
pick 4c849abaf945 mail/notmuch-emacs: Prevent a package name collision
pick 1614cc194cd5 net/tramp: Mark BROKEN and set EXPIRATION_DATE
pick 8591c9b3e6d8 UPDATING: Reiterate how to handle the emacs package name changes
squash e1d5f407046d UPDATING: Add entry for removal of major version from emacs packages
squash 156a7e7dae9e UPDATING: Set me as the author.  I should deal with any fallout.

# Rebase f76516f11076..156a7e7dae9e onto f76516f11076 (8 commands)
#
# Commands:
# C-c C-c  tell Git to make it happen
# C-c C-k  tell Git that you changed your mind, i.e. abort
# p        move point to previous line
# n        move point to next line
# M-p      move the commit at point up
# M-n      move the commit at point down
# SPC      show the commit at point in another buffer
# RET      show the commit at point in another buffer and select its window
# C-_      undo last change
# k        drop the commit at point
# y        insert a line for an arbitrary commit
# z        add noop action at point
# c        pick <commit> = use commit
# r, w     reword <commit> = use commit, but edit the commit message
# e, m     edit <commit> = use commit, but stop for amending
# s        squash <commit> = use commit, but meld into previous commit
# f        fixup [-C | -c] <commit> = like "squash" but keep only the previous
#                    commit's log message, unless -C is used, in which case
#                    keep only this commit's message; -c is same as -C but
#                    opens the editor
# x        exec <command> = run command (the rest of the line) using shell
# b        break = stop here (continue rebase later with 'git rebase --continue')
# l        label <label> = label current HEAD with a name
# t        reset <label> = reset HEAD to a label
# M M      merge [-C <commit> | -c <commit>] <label> [# <oneline>]
# .       create a merge commit using the original merge commit's
# .       message (or the oneline, if no original merge commit was
# .       specified); use -c <commit> to reword the commit message
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
```

**Comment by Jehops** [comment]
> The instructions are ambiguous. According to the previous steps the "commit message buffer" hasn't even been shown yet. You were in the "list of commits to be rebased buffer" and then typed `C-c C-c`.

> Are you trying to say that after doing that the "commit message buffer" is not created (which is very different from "once it existed but now it doesn't exist anymore").

Yes.  I am saying the `COMMIT_EDITMSG` buffer is not created when the repository is remote.  This happens with emacs built from the master branch (~ 10 days old, but the problem started earlier) and occurs if I run from an up-to-date clone of magit using `make emacs-Q`.  I just meant in the past (~ a few weeks or a month ago) the `COMMIT_EDITMSG` buffer was created as expected, but now it is not.

**Comment by tarsius** [comment]
Put like that it is not ambiguous anymore.

I suppose https://github.com/magit/with-editor/commit/8d52f933e50624c7bca3880f57297ac17ba4ac2d could have caused this. Try with `with-editor-3.0.5` instead of the development version.

**Comment by Jehops** [comment]
I'm still at a lose.  I tried `with-editor-3.0.5`, `with-editor-3.0.4`, reverting to Emacs 27.2 and tried different repositories on different hosts, but the result is the same.  No `COMMIT_EDITMSG` buffer is displayed.  I do see the correct `.git/COMMIT_EDITMSG` file is created.

**Comment by tarsius** [comment]
So it probably safe to say that this is not a regression in `with-editor`. I can't really think of something else that might have changed, but suspect it is something on your end.

**Comment by Jehops** [comment]
I have no ideas what it could be on my end given I was using `make emacs-Q` and/or all melpa-installed packages and my config moved out of the way.

In the past, I've always installed package from melpa.  I just tried another test.  I deleted the packages from melpa and installed OS packages.

```
dash.el-emacs_devel: 2.19.1_2
magit-devel-emacs_devel: 3.3.0.26_1
transient-emacs_devel: 0.3.7_1
with-editor-emacs_devel: 3.0.5_1
```
Now everything works as expected.  Maybe there is something different between the versions of other dependencies on melpa and these older versions of these emacs packages.  To confirm, you are able to do an interactive rebase, reword a commit and have the `COMMIT_EDITMSG` buffer displayed, all with the latest packages from melpa (and maybe to be sure, a recent emacs build from master)?

**Comment by tarsius** [comment]
I am able to do that yes (except I don't use Melpa, I use [`borg`](https://github.com/emacscollective/borg)). My emacs was build yesterday from master.

**Comment by dickmao** [comment]
At some point I had to turn on `magit-commit-diff-inhibit-same-window` for emacs-29.0.50.

I don't know what in my environment prevented `magit-display-buffer-traditional` from usurping the COMMIT_EDITMSG window (via `magit-commit-diff` via `server-switch-hook`), but whatever it was, it stopped.
